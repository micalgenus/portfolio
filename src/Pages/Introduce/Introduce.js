import React, { Component } from 'react';

export default class Introduce extends Component {
  render() {
    return (
      <div>
        <p>Introduce</p>
      </div>
    );
  }
}
