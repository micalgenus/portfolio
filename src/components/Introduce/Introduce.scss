.introduce-component {
  display: flex;
  justify-content: center;

  & > img {
    max-width: 200px;
    max-height: 200px;
  }
}
