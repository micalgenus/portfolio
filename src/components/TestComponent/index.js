import React from 'react';
import ReactDOM from 'react-dom';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

/**
 * @param {string} name Test name
 * @param {object} component Test component as <component /> object
 * @param {bool} snapshot if snapshot is true then call snapshot test in run
 * @param {function} callback run function after default test
 */
export const defaultTest = ({ name, component, snapshot = false, callback }) => {
  describe(name, () => {
    it('render test', () => {
      const div = document.createElement('div');
      ReactDOM.render(component, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    if (snapshot === true) {
      it('snapshot test', () => {
        const wrapper = shallow(component);
        expect(wrapper).toMatchSnapshot();
      });
    }

    if (typeof callback === 'function') callback.bind(this)();
  });
};

/**
 * @description Snapshot test alias
 * @param {string} name Test name
 * @param {object} component Test component as <component /> object
 * @param {function} callback run function after default test
 * @returns {undefined}
 */
export const snapshotTest = ({ name, component, callback }) => {
  if (!name || !component) return null;
  defaultTest({ name, component, snapshot: true, callback });
};
