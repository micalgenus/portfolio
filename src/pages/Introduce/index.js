import React from 'react';
import PageTemplate from '@/Pages';

export default class Introduce extends PageTemplate {
  render() {
    return (
      <div>
        <p>Introduce</p>
      </div>
    );
  }
}
