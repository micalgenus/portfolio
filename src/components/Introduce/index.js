import React, { Component } from 'react';

import { Icon } from 'semantic-ui-react';

import './Introduce.scss';

export default class Introduce extends Component {
  renderIcon = (href, icon, color = 'black') => {
    const linkRegex = /^http?(s):\/\//;

    return (
      <span>
        <a href={href} target={linkRegex.test(href) ? '_target' : '_self'}>
          <Icon name={icon} color={color} size="tiny" />
        </a>
      </span>
    );
  };

  renderLinkList = () => (
    <div className="link-list">
      {this.renderIcon('mailto:micalgenus@gmail.com', 'mail', 'black')}
      {this.renderIcon('https://github.com/micalgenus/', 'github', 'black')}
      {this.renderIcon('https://linkedin.com/in/gyeong-su-han/', 'linkedin', 'blue')}
      {this.renderIcon('https://micalgenus.github.io/', 'edit', 'grey')}
    </div>
  );
  render() {
    return (
      <div className="introduce-component">
        <div>
          <h1>한경수{this.renderLinkList()}</h1>
          <p>주로 Node.js를 사용하여 개발을 하고있는 신입 백엔드개발자 입니다. 현재 React와 CI/CD를 공부 하고 있습니다.</p>
        </div>
      </div>
    );
  }
}
