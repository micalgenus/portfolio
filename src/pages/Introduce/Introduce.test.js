import React from 'react';
import { snapshotTest } from '@/TestComponent';
import Introduce from './index';

snapshotTest({ name: '/Pages/Introduce', component: <Introduce /> });
