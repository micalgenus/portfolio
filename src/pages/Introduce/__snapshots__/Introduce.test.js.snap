// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Pages/Introduce snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Introduce />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Introduce
      </p>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Introduce",
      },
      "ref": null,
      "rendered": "Introduce",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Introduce
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Introduce",
        },
        "ref": null,
        "rendered": "Introduce",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
